### ISSUE TITLE
Colab demo requires multiple upgrades

### ISSUE THREAD

**Issue by Mayukhdeb** [DevoLearn/devolearn#69]
1. Devolearn requires `torch==1.7`, but I think it would also work with the latest versions of torch. Using the latets version of torch would make the installation much faster on google colab notebooks because the latest version of torch is already installed there.
2. The URL for the weights to the embryo segmentation model is most likely broken, throws an HTTP error.
3. arg `mode = 'cpu'` should be replaced with `device = 'cpu'`
4. optionally, we can also let the user decide the GPU ID like `device = 'cuda:0'`
5. move the `run tests` cell to the bottom

Note that this is a beginner friendly issue, all the changes are to be done on [this notebook](https://colab.research.google.com/github/DevoLearn/data-science-demos/blob/master/devolearn_docs/devolearn_quickstart.ipynb). Feel free to reply to this if you're willing to take this up.

**Comment by alon1samuel** [comment]
Hi, I've started to take a look on the issues, to help out :) 
Regarding 1 - Where is the colab notebook? 
a - I tried to go into Examples notebook - but it took me to [nbviewer.org notebook](https://nbviewer.org/github/DevoLearn/data-science-demos/blob/master/Networks/experiments_with_devolearn_node_maps.ipynb)
b - I'm seeing in [requirements.txt](https://github.com/DevoLearn/devolearn/blob/master/requirements.txt) that torch>=1.7 which is fine?

Continuing to check the rest

**Comment by sushmanthreddy** [comment]
@alon1samuel  https://colab.research.google.com/github/DevoLearn/data-science-demos/blob/master/devolearn_docs/devolearn_quickstart.ipynb this is link for the notebook and ,I was also going through the issue and found out main error was coming at "embryo_segmentor "actually ,I found out there was no file named embryo segmentor in the whole devolearn repo , I was still search for this thing, if u found out any leads ..pls let me knew

**Comment by Mayukhdeb** [comment]
Hi @sushmanthreddy and @alon1samuel 👋 
The colab notebook's source code can be found [here](https://github.com/DevoLearn/data-science-demos/blob/master/devolearn_docs/devolearn_quickstart.ipynb). Feel free to make a PR there with the fixed notebook.

**Comment by alon1samuel** [comment]
Thanks guys! I'll take a look @sushmanthreddy @Mayukhdeb

**Comment by Anantk2908** [comment]
I think, I have found the root of the problem. While running the "tests" cell. It throws an import error saying it `cannot import name 'cell_membrane_segmentor' from 'devolearn' (/usr/local/lib/python3.7/dist-packages/devolearn/__init__.py)` 

This error can be resolved by releasing a new package on pypi by the admins

![image](https://user-images.githubusercontent.com/56225125/184862897-7ee914d6-6131-4ac6-929e-11214ee3a204.png)

**Comment by alon1samuel** [comment]
So if you install the package from github it fixes the problem?
I mean if you do  `pip install git+https://github.com/DevoLearn/devolearn`  ... ?

**Comment by Anantk2908** [comment]
Yeah after trying the command the issue seems to resolved the only thing that remains is that embryo_segmentor is no longer a function that is used in the newer versions of devolearn and it's name has been changed to something else.
Can the admins update the same over here so it can be changed.

![image](https://user-images.githubusercontent.com/56225125/184867695-70dcae7d-28c0-4b50-962f-171de1d38d48.png)

**Comment by alon1samuel** [comment]
Thanks, I'll check it later. Hopefully create a PR soon

**Comment by sushmanthreddy** [comment]
after using 



> So if you install the package from github it fixes the problem?
> I mean if you do `pip install git+https://github.com/DevoLearn/devolearn` ... ?

no it doesn't resolve the issue actually the embryo_segmentor has been changed to cell_membrane_segmentor..when we try to import cell_membrane_segmentor it was throwing out an error at header files and while installing also it was showing an error..these were the errors ,which I have encountered..
<img width="1324" alt="image" src="https://user-images.githubusercontent.com/73489688/184951819-73ebb78d-edb8-4764-8149-a130b42de8d2.png">
<img width="1303" alt="image" src="https://user-images.githubusercontent.com/73489688/184951889-72f3a4fd-6485-45a6-9ea7-59c79ce06184.png">

**Comment by Anantk2908** [comment]
Hi @sushmanthreddy ,
If you delete the runtime and re-run the notebook the issue should be fixed. I did the same it worked for me.

**Comment by sushmanthreddy** [comment]
> Hi @sushmanthreddy ,
> If you delete the runtime and re-run the notebook the issue should be fixed. I did the same it worked for me.

thanks for helping out

**Comment by alon1samuel** [comment]
I've left a PR for the changes on devolearn side. 
Changes are still needed in the [data-science notebook](https://github.com/DevoLearn/data-science-demos)

**Comment by sushmanthreddy** [comment]
**fixed issues**
1 ." The URL for the weights to the embryo segmentation model is most likely broken, throws an HTTP error"  , I have tried to fix this issue .
2 . arg mode = 'cpu' should be replaced with device = 'cpu' , this has been fixed..

**Issues need  to  be fixed** :

1 .optionally, we can also let the user decide the GPU ID like device = 'cuda:0'
2 .move the run tests cell to the bottom..

**Comment by alon1samuel** [comment]
Thank you! Very nice work

**Comment by sushmanthreddy** [comment]
@alon1samuel 

use this code in run test cell...the path is inappropriate which u were using that is the reason it isn't working

```
import devolearn
os.path.dirname(devolearn.__file__) + "/" + "tests" + "/" + "test.py"
!python /content/devolearn/devolearn/test
```

 I have closed my pr because it has some incomplete work..and above code snippet works fine just path need to updated correctly
<img width="851" alt="image" src="https://user-images.githubusercontent.com/73489688/186249712-3b006532-d263-4b03-b03d-52bf08dc7cdc.png">


And I was encountering another error at installation place..at first code cell.I don't knew what it is! but it seems to look like this...if u find out same issue pls try too resolve it
<img width="1184" alt="image" src="https://user-images.githubusercontent.com/73489688/186250282-ddf7b7f8-3d39-4b60-9fff-9561ffd74911.png">

**Comment by alon1samuel** [comment]
Great! The tests run smoothly, I'll check on my notebook as well.

About your errors it's really weird about these dependencies. For example: I haven't seen tensorflow being needed for the installation. I'll slack you, maybe we can fix this in a huddle.

**Comment by alon1samuel** [comment]
I've fixed some other stuff in the notebook. 
Now the tests are failing since 1 file is not showing up. Even after installing devolearn from git by 

`pip install git+https://github.com/DevoLearn/devolearn.git@master`

I'm still getting a message - successfully installed devolearn-0.3.0 (which is weird)

And the file that is missing in the tests is "devolearn/tests/sample_data/images/nucleus_seg_sample.png":

![image](https://user-images.githubusercontent.com/45364499/188418205-5e904f1c-54bb-4d2f-a8dd-505cc31f55ed.png)


Anyone has an idea why id does that?

**Comment by alon1samuel** [comment]
I've ignored this in the end. Very strange anyway.

**Comment by alon1samuel** [comment]
I've created a [PR](https://github.com/DevoLearn/data-science-demos/pull/15) with recent changes to notebook on [repo](https://github.com/DevoLearn/data-science-demos)

I think the issue can be closed @Mayukhdeb as @Anantk2908 did most of the fixes there in the last MR
